sdk/farmer/src/metrics.rs

```diff
@@ -0,0 +1,271 @@
+use std::sync::atomic::AtomicU64;
+use std::time::Duration;
+
+use prometheus_client::metrics::counter::Counter;
+use prometheus_client::metrics::family::Family;
+use prometheus_client::metrics::histogram::{exponential_buckets, Histogram};
+use prometheus_client::registry::{Registry, Unit};
+use subspace_farmer::single_disk_farm::farming::ProvingResult;
+use subspace_farmer::single_disk_farm::{FarmingError, SingleDiskFarmId};
+
+#[derive(Debug, Clone)]
+pub(super) struct FarmerMetrics {
```

Comment:
This is fine for now, but we should de-duplicate this with monorepo and provide a nicer way to register metrics on disk farms and other things. Otherwise it is a lot of work to maintain the same code and easy to not add new metrics in Pulsar that are already present in `subspace-farmer`.